### ISSUE TITLE
Lacking VSCode functionality

### ISSUE THREAD

**Issue by OfekShilon** [microsoft/MIEngine#1368]
I'm going by [the docs](https://code.visualstudio.com/docs/cpp/natvis), submitting an issue where VSCode functionality is missing.
We (and probably many, many others) wish to use visualizers to debug hash maps and other standard data structures. Following [the documented example](https://code.visualstudio.com/docs/cpp/natvis), a few natvis constructs are needed for this:

1. CustomListItems,
2. Variable,
3. Exec,
4. Loop,
5. If

It seems at least the first item isn't implemented, I suspect the rest too. It would be very valuable to be able to watch various hash maps, sets and dictionaries.

**Comment by OfekShilon** [comment]
I suspect `AlternativeType` also isn't implemented

**Comment by OfekShilon** [comment]
Looking at [ExpandVisualzied](https://github.com/microsoft/MIEngine/blob/main/src/MIDebugEngine/Natvis.Impl/Natvis.cs#L553) I see it doesn't implement `CustomListItemsType`

**Comment by jkriegshauser** [comment]
I don't see any way to upvote on Github, but this is highly desirable. Also, the `<Intrinsic>` type as well.

**Comment by OfekShilon** [comment]
Ping. Any chance of doing something with this?

**Comment by OfekShilon** [comment]
@gregg-miskelly  Perhaps you can help prioritize this? This would make a world of difference in native debugging on VSCode.

**Comment by gregg-miskelly** [comment]
@OfekShilon I can't promise when we will be able to close more natvis gaps -- supporting natvis is difficult in MIEngine, as we need to do it on top of GDB commands instead of something built into the expression evaluator. But it would be useful if you could give us a prioritized list of the gaps (maybe the list at the top of this is already in your priority order)?

**Comment by OfekShilon** [comment]
@gregg-miskelly  I'd say:
 1. Loop,
 2. CustomListItems,
 3. Exec,
 4. Variable,
 5. If

**Comment by OfekShilon** [comment]
Is there any alternative? Can gdb pretty-printers be used to populate variable watch?

**Comment by gregg-miskelly** [comment]
Yes, if your launch options are configured to use pretty printing, then pretty printers can change the values seen in the variables window.

**Comment by SafariMonkey** [comment]
Is there a way to gracefully handle this lack of compatibility, without editing the XML to choose a version based on the capabilities of the debugger you're using?
